### DISCUSSION TITLE
Why not the same size with the original pdf when saving?

### DISCUSSION THREAD

**Discussion by tristone13th** [pymupdf/PyMuPDF#1821] [Q&A] [answered]
I use `doc.save("path-to-pdf")` to save a newly loaded pdf document, the original pdf is only 25MB, but the output is 50MB, why will this happen?

**Comment by JorjMcKie** [comment, marked as answer]
This may happen for a number of reasons.
First of all, even if you do nothing else than `fitz.open(file);fitz.save(newfile)`, this is **_not_** equivalent to simply doing `copy file -> newfile` with the means of your OS. Like every PDF package, MuPDF will reconstruct the PDF during save.
So you must provide the save options you need. E.g. if the input was a linear PDF, the output won't be if you don't request it.
Second, while there is a range of garbage selection and fine-grained compression options, MuPDF does not support certain other compression features like creating `/ObjStm` objects.
